src/main/java/org/generationcp/ibpworkbench/germplasm/pedigree/CreatePedigreeGraph.java

```diff
@@ -103,22 +140,34 @@ private void addNode(final GermplasmPedigreeTreeNode node, final int level) {
 		}
 
 		for (final GermplasmPedigreeTreeNode parent : node.getLinkedNodes()) {
+			final boolean isFemaleParent = parent.getGermplasm().getGid().equals(node.getGermplasm().getGpid1());
+			final boolean isDerivative = node.getGermplasm().getGpid1().equals(node.getGermplasm().getGpid2());
```

Comment:
Is it possible that a derivative germplasm has gpi1 not equal to gpid2?

Reply:
Hi @abatac , I don't think it's possible since it will be a cross if the parents are not the same.

Reply:
Based on this diagram, it's possible. You need to update the logic:

 
![Unknown](https://user-images.githubusercontent.com/5011207/106565397-b9e9a280-6569-11eb-8d7a-a3857c1d4027.png)